File: python/credit_check.py
def credit_check(str):
    reversed_list = reversed(list(str))
    is_valid = times_two(reversed_list)
    
    return 'The number is valid!' if is_valid else 'The number is invalid!'

def times_two(r_list):
    new_arr = []
    count = 0
    
    for i, n in enumerate(r_list):
        if i % 2 == 0:
            new_arr.append(int(n))
            count += int(n)
        else:
            if int(n) * 2 > 9:
                list_of_digits = list(str(int(n) * 2))
                digit_one = int(list_of_digits[0]) 
                digit_two = int(list_of_digits[1])
                new_arr.append(digit_one + digit_two )
                count += (digit_one + digit_two)
            else:
                new_arr.append(int(n) * 2)
                count += int(n) * 2
    return count % 10 == 0

File: js/creditCheck.js
function creditCheck(str) {
    arr_of_nums = str.split('')
    let isValid = timesTwoEveryOtherNum(arr_of_nums.reverse())

    return isValid
}

function timesTwoEveryOtherNum(arr) {
    let new_arr = []
    let count = 0
    for(i = 0; i < arr.length; i++) {
        if (i % 2 == 0) {
            let strToNum = parseInt(arr[i])
            new_arr.push(strToNum)
            count += strToNum
        }
        else {
            digitTimesTwo = parseInt(arr[i]) * 2

                if (digitTimesTwo > 9) {
                    let summedDigit = sumDigits(digitTimesTwo)
                    new_arr.push(summedDigit)
                    count += summedDigit
                } else {
                    new_arr.push(digitTimesTwo)
                    count += digitTimesTwo
                }
        }
    }
    return validatingCard(count)
}

function sumDigits(num) {
    let splittedNum = num.toString().split('')
    
    return parseInt(splittedNum[0]) + parseInt(splittedNum[1])
}

function validatingCard(num) {
    return num % 10 == 0 ? "The number is valid!" : "The number is invalid!"
}
module.exports = {creditCheck}